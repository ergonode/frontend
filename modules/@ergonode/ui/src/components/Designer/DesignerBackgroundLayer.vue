/*
 * Copyright Â© Ergonode Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div class="designer-background-layer">
        <slot />
        <template v-for="item in numberOfItems">
            <slot
                name="item"
                :column="getColumn(item - 1)"
                :row="getRow(item - 1)" />
        </template>
    </div>
</template>

<script>
export default {
    name: 'DesignerBackgroundLayer',
    props: {
        /**
         * Number of visible columns
         */
        columns: {
            type: Number,
            required: true,
        },
        /**
         * List of rows presented at Grid
         */
        rows: {
            type: Number,
            required: true,
        },
    },
    computed: {
        numberOfItems() {
            return this.columns * this.rows;
        },
    },
    methods: {
        getColumn(index) {
            return index % this.columns;
        },
        getRow(index) {
            return Math.floor(index / this.columns);
        },
    },
};
</script>

<style lang="scss" scoped>
    .designer-background-layer {
        position: absolute;
        top: 0;
        z-index: $Z_INDEX_LVL_0;
        display: grid;
        width: 100%;
        height: auto;
    }
</style>
