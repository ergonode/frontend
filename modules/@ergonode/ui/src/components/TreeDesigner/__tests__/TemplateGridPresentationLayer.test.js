/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import DesignerBackgroundLayer from '@UI/components/Designer/DesignerBackgroundLayer';
import {
    mount,
} from '@vue/test-utils';

describe('TreeDesigner/DesignerBackgroundLayer', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(DesignerBackgroundLayer, {
            propsData: {
                columns: 2,
                rows: 2,
                isHeader: false,
            },
        });
    });

    it('Component is named well', () => {
        expect(typeof DesignerBackgroundLayer.name).toBe('string');
        expect(DesignerBackgroundLayer.name).toEqual('DesignerBackgroundLayer');
    });

    it('Check number of items', () => {
        expect(DesignerBackgroundLayer.computed.numberOfItems.call({
            columns: 3,
            rows: 10,
        })).toBe(30);
    });

    it('Check default props', () => {
        expect(wrapper.vm.columns).toBe(2);
        expect(wrapper.vm.rows).toBe(2);
        expect(wrapper.vm.isHeader).toBe(false);
    });

    it('Check main classes witch default props', () => {
        const children = wrapper.findAll('.designer-background-layer > div');
        expect(wrapper.find('.designer-background-layer')).toBeTruthy();
        expect(wrapper.find('.template-grid-background-layer__border-top')).toBeTruthy();
        expect(children.length).toEqual(4);
        expect(wrapper).toMatchSnapshot();
    });

    // FIXME
    // it('Check main classes witch new props', () => {
    //     wrapper.setProps({
    //         columns: 3,
    //         isHeader: true,
    //     });
    //
    //     const children = wrapper.findAll('.designer-background-layer > div');
    //     expect(wrapper.find('.designer-background-layer').exists()).toBeTruthy();
    //     expect(wrapper.find('.template-grid-background-layer__border-top')).toBeFalsy();
    //     expect(children.length).toEqual(6);
    //     expect(wrapper).toMatchSnapshot();
    // });

    it.each([
        [
            1,
            false,
        ],
        [
            2,
            false,
        ],
        [
            3,
            true,
        ],
        [
            4,
            true,
        ],
    ])(
        'Set index (%s) get (%s)',
        (nr, bool) => {
            const check = wrapper.vm.markLastRow(nr);
            expect(check).toBe(bool);
        },
    );
});
