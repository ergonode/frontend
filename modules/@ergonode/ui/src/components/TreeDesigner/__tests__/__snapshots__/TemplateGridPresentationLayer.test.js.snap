// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeDesigner/DesignerBackgroundLayer Check main classes witch default props 1`] = `
<div class="designer-background-layer template-grid-background-layer__border-top">
  <div shadow-id="0" class="designer-background-item"></div>
  <div shadow-id="1" class="designer-background-item"></div>
  <div shadow-id="2" class="designer-background-item designer-background-item--last-row"></div>
  <div shadow-id="3" class="designer-background-item designer-background-item--last-row"></div>
</div>
`;
